tests/unit/s2n_protocol_preferences_test.c

```diff
@@ -160,6 +162,137 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* Test s2n_protocol_preference_read */
+    {
+        /* Safety checks */
+        {
+            struct s2n_stuffer stuffer = { 0 };
+            struct s2n_blob blob = { 0 };
+            EXPECT_ERROR_WITH_ERRNO(s2n_protocol_preference_read(NULL, &blob), S2N_ERR_NULL);
+            EXPECT_ERROR_WITH_ERRNO(s2n_protocol_preference_read(&stuffer, NULL), S2N_ERR_NULL);
+        }
+
+        /* Read malformed value */
+        {
+            struct s2n_stuffer input = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, 0));
+            EXPECT_SUCCESS(s2n_stuffer_write_uint8(&input, sizeof(protocol1) + 10));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&input, protocol1, sizeof(protocol1)));
+
+            struct s2n_blob result = { 0 };
+            EXPECT_ERROR(s2n_protocol_preference_read(&input, &result));
```

Comment:
Can we be specific and assert the error_no using `EXPECT_ERROR_WITH_ERRNO` here so that we know the function is error-ing at the right place due to the wrong size?